src/models/message_status.ts

```diff
@@ -57,12 +64,26 @@ export const getMessageStatusUpdater = (
   messageId: NonEmptyString
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 ): MessageStatusUpdater => status =>
-  messageStatusModel.upsert({
-    kind: "INewMessageStatus",
-    messageId,
-    status,
-    updatedAt: new Date()
-  });
+  pipe(
+    messageStatusModel.findLastVersionByModelId([messageId]),
+    TE.map(
+      O.getOrElse(() => ({
+        messageId,
+        status,
```

Comment:
Maybe `status` should be overridden in any case if a message-status is already present

Reply:
ops, my ad. Done in [2f43605](https://github.com/pagopa/io-functions-commons/pull/239/commits/2f43605ca4cbd7ca67cf37324614389ff6a4aa5e)